packages/synapse-react-client/src/components/ChallengeSubmission/SubmissionDirectoryList.tsx

```diff
@@ -0,0 +1,376 @@
+          )
+        }
+      }
+      setCanSubmit(true)
+      setErrorMessage(undefined)
+    }
+  }
+
+  const columns: GridColDef[] = [
+    {
+      field: 'radiobutton',
+      headerName: '',
+      width: 25,
+      sortable: false,
+      filterable: false,
+      hideable: false,
+      disableColumnMenu: true,
+      renderCell: params => {
+        return (
+          <RadioOption
+            value={params.id}
+            currentValue={selectedItem?.id}
+            onChange={selectedItemId => {
+              entityChangeHandler(selectedItemId as string)
+            }}
+            label=""
+            groupId="radiogroup"
+            style={{ marginBottom: '16px' }}
+          />
+        )
+      },
+    },
+    {
+      field: 'name',
+      headerName:
+        entityType === EntityType.DOCKER_REPO
+          ? 'Docker Repository'
+          : 'File Name',
+      flex: 1,
+      filterable: false,
+      hideable: false,
+      disableColumnMenu: true,
+      renderCell: (params: GridCellParams<Entity, SubmissionDirectoryRow>) => (
+        <Link
+          to={{
+            pathname: `${getEndpoint(
+              BackendDestinationEnum.PORTAL_ENDPOINT,
+            )}#!Synapse:${params.row.id}`,
+          }}
+          target="_blank"
+        >
+          {params.row.name}
+        </Link>
+      ),
+    },
+    {
+      field: 'modifiedOn',
+      headerName: 'Updated On',
+      width: 100,
+      filterable: false,
+      hideable: false,
+      disableColumnMenu: true,
+    },
+    {
+      field: 'id',
+      headerName: 'ID',
+      width: 200,
+      filterable: false,
+      hideable: false,
+      disableColumnMenu: true,
+    },
+  ]
+
+  const getRows = (entities: EntityItem[]) => {
+    const newRows: SubmissionDirectoryRow[] = []
+    entities.forEach(entity => {
+      newRows.push({
+        id: entity.id!,
+        name:
+          entityType === EntityType.DOCKER_REPO
+            ? entity.repositoryName ?? entity.name
+            : entity.name,
+        modifiedOn: formatDate(dayjs(entity.modifiedOn), 'MM/DD/YY'),
+      })
+    })
+    return newRows
+  }
+
+  const handlePageChange = (newPageNum: number) => {
+    const lastIndexNeeded = Math.min(
+      headerResults?.totalChildCount ?? 0,
+      (newPageNum + 1) * PER_PAGE,
+    )
+    if (lastIndexNeeded > fetchedHeaders.length) {
+      setFetchNextPage(true)
+    }
+    setPage(newPageNum)
+  }
+
+  const itemSelectedHandler = () => {
+    onItemSelected(selectedItem!)
+  }
+
+  const handleUpload = (response: UploadCallbackResp) => {
+    if (response.success && response.resp) {
+      // Create Entity
+      const { fileHandleId, fileName } = response.resp
+      const newFileEntity: FileEntity = {
+        parentId: challengeProjectId,
+        name: fileName,
+        concreteType: FILE_ENTITY_CONCRETE_TYPE_VALUE,
+        dataFileHandleId: fileHandleId,
+      }
+      SynapseClient.createEntity(newFileEntity, accessToken)
```

Comment:
This file may already exist and they may want to overwrite the value (create a new version).  Please try to find child  using https://rest-docs.synapse.org/rest/POST/entity/child.html
(either before you attempt to create, or after you get receive a 409 status code error when trying to create the entity and you run into the conflict).  If child exists with this name, then you should just update the FileEntity dataFileHandleId and update the entity in the backend (preferably after a confirmation dialog, but you can ignore for now if that saves time).